src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -2972,6 +2973,21 @@ int MPL_ze_ipc_handle_create(const void *ptr, MPL_gpu_device_attr * ptr_attr, in
             MPL_ze_gem_hash_entry_t *entry = NULL;
             HASH_FIND_PTR(gem_hash, &ptr, entry);
 
+            /* invalid entry */
+            if (entry && entry->mem_id != mem_id) {
+                HASH_DEL(gem_hash, entry);
+                /* close GEM handle */
```

Comment:
@zhenggb72 should we call `MPL_gpu_ipc_handle_destroy` here?

Reply:
I think `MPL_gpu_ipc_handle_destroy` should be called in https://github.com/pmodels/mpich/blob/ea45c27f4267c68e57ce8b635ddeab52f2765bb5/src/mpl/src/gpu/mpl_gpu_ze.c#L1685-L1688. 

`MPL_gpu_ipc_handle_destroy` takes care of `ipc_cache_tracked` and `gem_hash` so the checks here shouldn't be needed.

Reply:
I tested it, the test pass. So I think we can call destroy instead.

Reply:
With the fixup patch the `generic` and `specialized` cache configuration work, but `disabled` still fails.

Reply:
disabled mode worked as I tested yesterday with the branch

Reply:
Yup. We need to add this hunk back. I will force push again shortly.